downloads/downloads-impl/src/main/java/com/duckduckgo/downloads/impl/DefaultFileDownloadNotificationManager.kt

```diff
@@ -21,69 +21,112 @@ import android.app.PendingIntent
 import android.content.Context
 import android.content.Intent
 import android.net.Uri
-import android.os.Handler
-import android.os.Looper
 import androidx.annotation.AnyThread
 import androidx.core.app.NotificationCompat
+import androidx.core.content.FileProvider
+import com.duckduckgo.appbuildconfig.api.AppBuildConfig
 import com.duckduckgo.di.scopes.AppScope
 import com.duckduckgo.downloads.api.FileDownloadNotificationManager
 import com.squareup.anvil.annotations.ContributesBinding
 import dagger.SingleInstanceIn
+import java.io.File
 import javax.inject.Inject
 
 @AnyThread
 @ContributesBinding(AppScope::class)
 @SingleInstanceIn(AppScope::class)
 class DefaultFileDownloadNotificationManager @Inject constructor(
     private val notificationManager: NotificationManager,
-    private val applicationContext: Context
+    private val applicationContext: Context,
+    private val appBuildConfig: AppBuildConfig,
 ) : FileDownloadNotificationManager {
 
-    override fun showDownloadInProgressNotification() {
-        mainThreadHandler().post {
-            val notification = NotificationCompat.Builder(applicationContext, FileDownloadNotificationChannelType.FILE_DOWNLOADING.id)
-                .setContentTitle(applicationContext.getString(R.string.downloadInProgress))
-                .setSmallIcon(R.drawable.ic_file_download_white_24dp)
-                .build()
+    @AnyThread
+    override fun showDownloadInProgressNotification(downloadId: Long, filename: String, progress: Int) {
+        val pendingIntent = PendingIntent.getBroadcast(
+            applicationContext,
+            1234,
+            FileDownloadNotificationActionReceiver.cancelDownloadIntent(downloadId),
+            PendingIntent.FLAG_UPDATE_CURRENT
+        )
+        val notification = NotificationCompat.Builder(applicationContext, FileDownloadNotificationChannelType.FILE_DOWNLOADING.id)
```

Comment:
You may want to add here `.setOngoing(true)` to disable the action of swiping the notification and achieve the same behaviour as the DownloadManager. At the moment you can swipe the notification and it appears again.